packages/design-picker/src/components/index.tsx

```diff
@@ -241,6 +253,8 @@ const DesignPicker: React.FC< DesignPickerProps > = ( {
 		? designs
 		: filterDesignsByCategory( designs, selectedCategory );
 
+	filteredDesigns.sort( sortDesigns );
```

Comment:
IMO, sorting in client/signup/steps/design-picker/index.jsx:L86 might be better as we don't need this on gutenboarding.

Reply:
I added it here because it's part of the `showFirst` logic. Do you think it'd be better to have the design step be responsible for adding Blank Canvas to every category?
It's possible, but we'd need to artificially add every category to the Blank Canvas theme, and the design step doesn't really have any concept of categories at the moment (i.e. `gatherCategories` is in the DP component)

Reply:
Oh I don't notice that you've added the `showFirst` to the type definition of `Design`. But I still think we should avoid sorting at rendering. It's okay for now as the size of `filteredDesigns` is small. However, the `sort` function might cause the side effect easily because it mutate the original array. If we need to use it here, I prefer to clone the value first.
```suggestion
	const sortedDesign = filteredDesigns.slice().sort( sortDesigns );
```

Reply:
I hadn't noticed how I was mutating the prop that was passed in if `! showCategoryFilter`! That feels like poor form and it should be cloned. Perhaps I'll add a `useMemo()` too since I don't need to sort for _every_ render.

Reply:
[Turns out `.slice` is the most consistently fast method of cloning on different browsers](https://perf.link/#eyJpZCI6Im41dXZhZ3Z6aXdxIiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6ImNvbnN0IGRhdGEgPSBbXVxuZm9yKGxldCBpID0gMDsgaSA8IDEwMDAwOyBpKyspIHsgZGF0YS5wdXNoKE9iamVjdC5jcmVhdGUoe30pKSB9IiwidGVzdHMiOlt7Im5hbWUiOiJzbGljZSIsImNvZGUiOiJjb25zdCByZXN1bHQgPSBkYXRhLnNsaWNlKDApIiwicnVucyI6WzEzMzc1LDE0MTI1LDE2MDAwLDEzMTI1LDExNTAwLDEzODc1LDE0MTI1LDE0MDAwLDE0MjUwLDEzNTAwLDk3NTAsMTIyNTAsMTgyNTAsMTQ2MjUsMTMzNzUsMTM2MjUsMTQwMDAsMTMzNzUsMTM2MjUsNDUwMCwxMzM3NSwxMzM3NSwxMzUwMCwxMzYyNSwyMTg3NSwxNDAwMCwyMTI1MCwxODEyNSwxMzc1MCwyMDEyNSwxMzM3NSwxMDUwMCwxMTYyNSwyNzM3NSwxMjI1MCwxMzM3NSwxMzg3NSwxMzc1MCwxOTAwMCwxMzM3NSwxMzUwMCwxMzM3NSwxMzc1MCwxMzc1MCwxMzc1MCwxMzg3NSwxNDAwMCwxMzYyNSwxMTI1MCwxNTM3NSwxNjc1MCwxMzc1MCwxMDg3NSwxMzI1MCwxMjUwMCwxNDM3NSwxMDM3NSwxMzM3NSwyMjM3NSwxODI1MCwxMjYyNSwxNDAwMCwxMzAwMCwxNDAwMCw5MDAwLDEzNTAwLDIxMDAwLDEzNTAwLDE3MTI1LDEwMjUwLDEzMzc1LDEyMjUwLDEzNzUwLDEyMjUwLDE2NjI1LDExMjUwLDE5MDAwLDEzMzc1LDEzNjI1LDExNjI1LDEzMTI1LDEzNzUwLDEzNzUwLDEzNTAwLDEzMzc1LDEzMzc1LDEzNjI1LDEzNTAwLDExODc1LDUzNzUsMTM2MjUsMTMzNzUsMjA4NzUsMTM1MDAsMTM1MDAsMTc2MjUsMTM2MjUsMTMzNzUsMTE4NzUsMTAzNzVdLCJvcHMiOjE0MDEwfSx7Im5hbWUiOiJzcHJlYWQiLCJjb2RlIjoiY29uc3QgcmVzdWx0ID0gWy4uLmRhdGFdIiwicnVucyI6WzUwMCwxMzc1LDEwMDAsMTc1MCwxMzc1LDEyNSwxNTAwLDIzNzUsMTg3NSwyMzc1LDEzNzUsMTYyNSw2MjUsMTUwMCwxNTAwLDEzNzUsMTEyNSwyMDAwLDEyNSwxODc1LDE4NzUsNzUwLDM3NSw2MjUsMzEyNSwxODc1LDEzNzUsMTM3NSwyNTAsMTM3NSwxNTAwLDEzNzUsMTM3NSwyNTAwLDEzNzUsMTM3NSwxMzc1LDE4NzUsMzc1LDM3NSwyNTAwLDYyNSw4NzUsNTAwLDEzNzUsMjYyNSwxMzc1LDUwMCwyMzc1LDEzNzUsMTM3NSwxNTAwLDEzNzUsMTM3NSwxNTAwLDE2MjUsMTM3NSwzNzUsNzUwLDEzNzUsNTAwLDEzNzUsMTM3NSwxMzc1LDEzNzUsMTAwMCwxMjUsMTI1LDEwMDAsMTM3NSwxMzc1LDE2MjUsMTI1LDEzNzUsMTM3NSwxNTAwLDEzNzUsMTAwMCwxMjUwLDExMjUsMTI1LDEzNzUsMTI1LDE4NzUsMTI1LDEwMDAsMTEyNSwxMjUsMTM3NSwyMDAwLDEwMDAsMTI1LDI1MCwxMTI1LDEzNzUsMTM3NSwxMzc1LDEyNTAsMTM3NSwxNTAwXSwib3BzIjoxMjIzfSx7Im5hbWUiOiIuZnJvbSIsImNvZGUiOiJjb25zdCByZXN1bHQgPSBBcnJheS5mcm9tKGRhdGEpIiwicnVucyI6WzUwMCw4NzUsNTAwLDc1MCw4NzUsODc1LDEyNTAsODc1LDEyNTAsNzUwLDc1MCwzNzUsODc1LDM3NSwxNTAwLDEwMDAsMTI1LDc1MCwyMzc1LDEwMDAsMTAwMCwyMDAwLDEwMDAsNTAwLDg3NSwyMDAwLDg3NSwxMjUwLDE1MDAsNzUwLDI1MCwxMjUwLDg3NSwyMDAwLDIwMDAsNjI1LDYyNSw3NTAsNTAwLDc1MCwxMDAwLDc1MCw3NTAsNzUwLDEwMDAsNzUwLDg3NSw3NTAsODc1LDc1MCwxMjUwLDg3NSwxMDAwLDg3NSw3NTAsODc1LDc1MCwyMDAwLDc1MCw4NzUsMTAwMCw3NTAsNzUwLDg3NSw4NzUsNzUwLDI1MCw3NTAsNjI1LDEyNTAsODc1LDg3NSw4NzUsODc1LDc1MCw4NzUsMjUwLDEwMDAsODc1LDUwMCwxMDAwLDg3NSw3NTAsNjI1LDg3NSw4NzUsMTEyNSwzNzUsODc1LDg3NSw3NTAsMjUwLDI1MCw1MDAsNzUwLDM3NSw3NTAsMzc1LDM3NSw4NzVdLCJvcHMiOjg2M31dLCJ1cGRhdGVkIjoiMjAyMS0xMS0xNVQxOTo0MToyMS42NDRaIn0%3D). TIL